api/filters.go

```diff
@@ -402,10 +408,18 @@ func (api *API) validateDimensions(filterDims []model.Dimension, dims []dataset.
 	dimensions := make(map[string]string)
 	for _, d := range dims {
 		dimensions[d.Name] = d.ID
+
 	}
 	var incorrect []string
 	for _, fd := range filterDims {
 		if _, ok := dimensions[fd.Name]; !ok {
+			// if this is a cantabular multivariate table, then you
+			// should be free to create whatever dimension that you
+			// want to add, provided that they exist. which is checked
+			// in dimension option validation.
+			if datasetType == "cantabular_multivariate_table" {
```

Comment:
IIRC the dimension options check only performs the test if there are specific options chosen. If this is left blank(where users would want all options) then no check will be done. Another solution is likely needed

Reply:
Also, this check could be moved out of the loop just to return straight away here as nothing else will be done